src/NHibernate/Mapping/ByCode/FakeModelExplicitDeclarationsHolder.cs

```diff
@@ -15,6 +15,8 @@ public class FakeModelExplicitDeclarationsHolder : IModelExplicitDeclarationsHol
 		private readonly IEnumerable<MemberInfo> idBags = Enumerable.Empty<MemberInfo>();
 		private readonly IEnumerable<MemberInfo> lists = Enumerable.Empty<MemberInfo>();
 		private readonly IEnumerable<MemberInfo> manyToManyRelations = Enumerable.Empty<MemberInfo>();
+		private readonly IEnumerable<MemberInfo> entityManyToManyRelations = Enumerable.Empty<MemberInfo>();
+		private readonly IEnumerable<MemberInfo> keyManyToManyRelations = Enumerable.Empty<MemberInfo>();
```

Comment:
I mean, you do not need the ManyToManyRelations property which returns the union at all.